.github/workflows/functional-test.yaml

```diff
@@ -23,7 +23,11 @@ on:
     - cron: "30 0,12 * * 0,6"
   # Dispatch on external events
   repository_dispatch:
-    types: [functional-tests, de-functional-test]
+    types: [de-functional-test]
+  workflow_run:
+    workflows: ['Functional test approval using environments']
+    types:
+      - completed
```

Comment:
It's kinda nice that with this separation between approval and run, we can still use the scheduled runs and support repository dispatch.